File: src/net/GOOM.TR.MyRetail.NET/GOOM.TR.MyRetail.NET/Services/ProductService.cs
using GOOM.TR.MyRetail.NET.Models;
using GOOM.TR.MyRetail.NET.Repos;
using Microsoft.Extensions.Logging;
using System;
using System.Threading.Tasks;

namespace GOOM.TR.MyRetail.NET.Services
{
    public interface IProductService
    {
        Task<Product> GetProductAsync(long id);
        Task<Price> SetProductCurrentPriceAsync(long id, Price price);
        Task<bool> DeleteProductCurrentPriceAsync(long id);
    }

    public class ProductService : IProductService
    {
        private readonly ILogger<ProductService> _log;
        private readonly IProductRepo _productRepo;
        private readonly IPriceRepo _priceRepo;

        public ProductService(ILogger<ProductService> log, IProductRepo productRepo, IPriceRepo priceRepo)
        {
            _log = log;
            _productRepo = productRepo;
            _priceRepo = priceRepo;
        }

        public async Task<Product> GetProductAsync(long id)
        {
            try
            {
                var product = await _productRepo.GetProductAsync(id);
                if (product == null) return product;

                product.CurrentPrice = await _priceRepo.GetPriceByProductIdAsync(id);
                return product;
            } 
            catch (Exception ex)
            {
                _log.LogError(ex, ex.Message);
            }
            return null;
        }

        public async Task<Price> SetProductCurrentPriceAsync(long id, Price price)
        {
            try
            {
                return await _priceRepo.SetProductPriceAsync(id, price);
            }
            catch (Exception ex)
            {
                _log.LogError(ex.Message, ex);
            }
            return null;
        }

        public async Task<bool> DeleteProductCurrentPriceAsync(long id)
        {
            try
            {
                return await _priceRepo.DeleteProductPriceAsync(id);
            }
            catch (Exception ex)
            {
                _log.LogError(ex.Message, ex);
            }
            return false;
        }
    }
}

File: src/java/src/main/java/com/goom/tr/myretail/services/ProductService.java
package com.goom.tr.myretail.services;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.goom.tr.myretail.models.*;
import com.goom.tr.myretail.repos.*;
import com.goom.tr.myretail.repos.redsky.models.ProductResponse;

public class ProductService implements IProductService {

	private Logger Log = LoggerFactory.getLogger(ProductService.class);

	private final IProductRepo _productRepo;
	private final IPriceRepo _priceRepo;

	public ProductService(IProductRepo productRepo, IPriceRepo priceRepo) {
		_productRepo = productRepo;
		_priceRepo = priceRepo;
	}

	@Override
	public Product GetProduct(long id) {

		try {
			Product product = _productRepo.GetProduct(id);
			if (product == null)
				return product;

			product.CurrentPrice = _priceRepo.GetPriceByProductId(id);
			return product;
		} catch (Exception ex) {
			Log.error(ex.getMessage(), ex);
		}
		return null;
	}

	@Override
	public Price SetProductCurrentPrice(long id, Price price) {
		try {
			return _priceRepo.SetProductPrice(id, price);
		} catch (Exception ex) {
			Log.error(ex.getMessage(), ex);
		}
		return null;
	}

	@Override
	public boolean DeleteProductCurrentPrice(long id) {
		try {
			return _priceRepo.DeleteProductPrice(id);
		} catch (Exception ex) {
			Log.error(ex.getMessage(), ex);
		}
		return false;
	}

}
